### PR TITLE
 Improve Contentful-related test coverage - part 2 of n

### PR DESCRIPTION
**Pull request by stevejalim**
## Description

Backfilling more tests for the contentful integration. These are having to be written based on viewable code paths, so aren't going to cover every edge case, but they're a building block.

More to come soon - hopefully just one PR's worth.

## Issue / Bugzilla link

Contributes to #10639

## Testing

Unit tests passing and coverage going (slightly) up is enough,  I think.

### CODE REVIEW

File: bedrock/contentful/api.py [hunk 1]
```diff
@@ -257,6 +257,8 @@ def render(self, node):
         data_cta = ""
 
         # add referral info to links to other mozilla properties
+        # TODO: can we make this just if url.netloc == "mozilla.org" or are we expecting
```
Comment (pmac) [inline bedrock/contentful/api.py:260]: Right. This block is for linking to other mozilla.org subdomains, and the one below is for other non mozilla.org domains. We could perhaps update this to deal with `*.firefox.com` domains as well.

File: bedrock/contentful/tests/test_contentful_api.py [hunk 2]
```diff
@@ -466,3 +478,171 @@ def test__only_child():
...
+def test__StrongRenderer():
+    assert StrongRenderer().render(mock_simple_node) == "<strong>foo</strong>"
+
+
+def test__EmphasisRenderer():
+    assert EmphasisRenderer().render(mock_simple_node) == "<em>foo</em>"
+
+
+@patch("bedrock.contentful.api.get_current_request")
+def test__LinkRenderer__mozilla_link(mock_get_current_request):
```
Comment (pmac) [inline bedrock/contentful/tests/test_contentful_api.py:502]: why double underscores? Just curious.

File: bedrock/contentful/tests/test_contentful_api.py [hunk 3]
```diff
@@ -466,3 +478,171 @@ def test__only_child():
...
+
+    mock_request = Mock()
+    mock_request.page_info = {"utm_campaign": "TEST"}
+    mock_get_current_request.return_value = mock_request
+    mozilla_mock_hyperlink_node = deepcopy(mock_hyperlink_node)
+    mozilla_mock_hyperlink_node["data"]["uri"] = "https://mozilla.org/test/page/"
```
Comment (robhudson) [inline bedrock/contentful/tests/test_contentful_api.py:508]: Just curious if the actual code uses `mozilla.org` or `www.mozilla.org`? Because the first redirect you'll get is if you go to the subdomain-less URL.

File: bedrock/contentful/tests/test_templatetags_helpers.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        "img",
+        "li",
+        "ol",
+        "p",
+        "small",
+        "span",
+        "strike",
+        "strong",
+        "ul",
+        "hr",
```
Comment (robhudson) [inline bedrock/contentful/tests/test_templatetags_helpers.py:105]: Just happened to noticed these look sorted except for the `hr`.

File: bedrock/contentful/tests/test_contentful_api.py [hunk 5]
```diff
@@ -466,3 +478,171 @@ def test__only_child():
...
+                "list-item": ListItemRenderer,
+            }
+        ).render(mock_list_node)
+        == "<ol class='mzp-u-list-styled'><li>foo</li></ol>"
+    )
+
+
+@pytest.mark.skip("TODO")
+def test__LiRenderer():
+    assert False, "WRITE ME"
```
Comment (robhudson) [inline bedrock/contentful/tests/test_contentful_api.py:572]: ⚠️

**Review by robhudson — APPROVED** [review]
A couple comments otherwise looks nice

**Review by pmac — APPROVED** [review]
Nice 🎸

File: bedrock/contentful/tests/test_contentful_api.py (hunk 5 above)
Comment (stevejalim) [reply bedrock/contentful/tests/test_contentful_api.py:572]: Yep - that's getting addressed in the next batch. Thanks for the laser-eyes

File: bedrock/contentful/tests/test_templatetags_helpers.py (hunk 4 above)
Comment (stevejalim) [reply bedrock/contentful/tests/test_templatetags_helpers.py:105]: Nice spot - will amend in the next batch of work

File: bedrock/contentful/tests/test_contentful_api.py (hunk 2 above)
Comment (stevejalim) [reply bedrock/contentful/tests/test_contentful_api.py:502]: First dunder is a typo (will fix), the second is how I separate the thing being tested from the aspect/case am testing.

File: bedrock/contentful/tests/test_contentful_api.py (hunk 3 above)
Comment (stevejalim) [reply bedrock/contentful/tests/test_contentful_api.py:508]: The actual code has this
```
if "mozilla.org" in url.netloc and url.netloc != "www.mozilla.org":
```
and - you're right - it will 302 to www, so I'm _assuming_ this code is here for links from, say, blog.m.o. 

Will update the test with that assumption and add a note, for future selves

File: bedrock/contentful/api.py (hunk 1 above)
Comment (stevejalim) [reply bedrock/contentful/api.py:260]: @pmac can you elaborate on this (DMs are fine) - i'll add the changes to the next block of contentful tests
